### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by zegnus**
### Description

This Pull Request updates the dependencies:
- gradle tools from v1.2.3 -> v2.3.0 [changelog](https://developer.android.com/studio/releases/gradle-plugin.html#)
- gradle from v2.2 -> v3.3 [changelog](https://docs.gradle.org/3.3/release-notes.html)
- android build tools from 23.0.2 -> 25.0.2 [changelog](https://developer.android.com/studio/releases/build-tools.html#)
- compile sdk version from 23 -> 25
- appcompat-v7 from 23.1.1 -> 25.2.0 [changelog](https://developer.android.com/topic/libraries/support-library/revisions.html)
- novoda bintray v0.2.4 -> 0.4.0 [changelog](https://github.com/novoda/bintray-release/releases/tag/0.4.0)

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -15,12 +15,12 @@ apply plugin: 'com.android.library'
...
 android {
-    compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    compileSdkVersion 25
+    buildToolsVersion "25.0.2"
 
     defaultConfig {
         minSdkVersion 14
-        targetSdkVersion 23
+        targetSdkVersion 25
```
Comment (rock3r) [inline android/build.gradle]: Have you checked this doesn't impact anything in the library functionality?

Comment (tasomaniac) [inline android/build.gradle]: It is actually removed in PR #76 
An explanation is given there. Increasing this will increase it for every client.

**Comment by tasomaniac** [comment]
Isn't it unfortunate that we have #76 and #77 conflicting very hard? I would suggest to wait for #76 to be merged. I already approved but was waiting for more 👀

**Comment by ataulm** [comment]
ah darn, I thought this was already merged.

@zegnus @tasomaniac @rock3r do you want me to update my one with these version numbers?

**Comment by ataulm** [comment]
@zegnus I merged master into this:

- the gradle wrapper was as you did it
- the target sdk version is no longer specified
- all other version updates are in dependencies.gradle

**Pull request merged**